plugins/modules/user_role_workflow_manager.py

```diff
@@ -1024,6 +1033,44 @@ def validate_role_parameters(self, role_key, params_list, role_config, role_para
                                              "DEBUG")
                                     self.validate_string_parameter(param, inventory_management[param], error_messages)
 
+    def identify_invalid_params(self, params, mismatches):
+        """
+        Identify and collect invalid parameters from a dictionary or list based on allowed parameters.
+        Args:
+            - params (dict): The dictionary of parameters to be checked. Nested dictionaries or lists are supported.
```

Comment:
Looks like you check both dict and list..

`        params (dict | list): The dictionary or list of parameters to be checked. Nested dictionaries or lists are supported.`

Reply:
Updated the code accordingly.